core/src/main/java/org/jboss/logmanager/LogManager.java

```diff
@@ -115,17 +115,23 @@ private void doConfigure(InputStream inputStream) {
             synchronized (configuratorRef) {
                 configurator = configuratorRef.get();
                 if (configurator == null) {
-                    final ServiceLoader<LogContextConfigurator> serviceLoader = ServiceLoader.load(LogContextConfigurator.class);
-                    final Iterator<LogContextConfigurator> iterator = serviceLoader.iterator();
+                    int best = Integer.MAX_VALUE;
+                    ConfiguratorFactory factory = null;
+                    final ServiceLoader<ConfiguratorFactory> serviceLoader = ServiceLoader.load(ConfiguratorFactory.class);
+                    final Iterator<ConfiguratorFactory> iterator = serviceLoader.iterator();
                     List<Throwable> problems = null;
                     for (;;) try {
                         if (! iterator.hasNext()) break;
-                        configurator = iterator.next();
-                        break;
+                        final ConfiguratorFactory f = iterator.next();
+                        if (f.priority() < best || factory == null) {
```

Comment:
I guess lower priority is better.  This should be documented up in the factory interface.

Reply:
Yeah. I'm not sure if higher or lower should be preferred, but I'll document either way.